//
// Aspia Project
// Copyright (C) 2020 Dmitry Chapyshev <dmitry@aspia.ru>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/>.
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

import "router_common.proto";

package proto;

message RelayKeyPool
{
    // A pool of one time keys.
    repeated RelayKey key = 1;
}

message RelayKeyPoolRequest
{
    // The number of keys requested.
    uint32 pool_size = 1;
}

// Sent from proxy to router.
message RelayToRouter
{
    RelayKeyPool key_pool = 1;
}

// Sent from router to proxy.
message RouterToRelay
{
    RelayKeyPoolRequest key_pool_request = 1;
}
